calculator/main.py

```diff
@@ -25,9 +26,21 @@ def perform_operation(calc, operation, params):
     :param operation: String with the operation name. ie: 'add'
     :param params: Tuple containing the list of nums to operate with.
                    ie: (1, 2, 3, 4.5, -2)
+                   
     """
-    pass
-
+    if operation not in calc['operations'].keys():
```

Comment:
You don't need to use `keys()`. This is going to make it less efficient. Remind me to explain why on Saturday please.